### ISSUE TITLE
Support Fingerprint Login

### ISSUE THREAD

**Issue by cancan101** [auth0/Lock.Android#356]
[This seems to indicate](https://auth0.com/learn/fingerprint-authentication/) fingerprint support on Android, but I don't see it documented / used. 

See also: https://developer.android.com/about/versions/marshmallow/android-6.0.html#fingerprint-authentication

**Comment by hzalaz** [comment]
It only talks about the TouchID support we have for iOS plus a very basic background on fingerprint. On Android we will evaluate to add it when more people has a device with a reader and Android M at least.

**Comment by cancan101** [comment]
So why not keep this issue open until then? Also it looks [like 19%+ of users are now on M](https://developer.android.com/about/dashboards/index.html#Platform).

**Comment by hzalaz** [comment]
We will consider it when at least 30% of users are in api level 23+ and it's not a common thing to have an Android device with fingerprint reader. It's closed since it's on our internal list of things to research/consider in the future.

**Comment by steelbrain** [comment]
Hi @hazalaz

Current statistics indicate Android M usage is `31.2%` and if we combine M+ it becomes `36.2%`
<img src="https://cloud.githubusercontent.com/assets/4278113/25536606/e0599120-2c54-11e7-9156-28bc20c5be15.png">

Can you reconsider supporting this?

**Comment by hzalaz** [comment]
@steelbrain I'd not add something in the same way iOS has now. We only think it makes sense if you want to protect the obtained tokens or as an enhanced remember me but that could easily be outside lock. 

We will review the touchid passwordless flow at one point this year to make it better suited for Android but its not something its on our immediate pipeline

**Comment by rubixhacker** [comment]
Is there any updates to this? M+ is now 84.9% of active devices

**Comment by hon13002** [comment]
> Is there any updates to this? M+ is now 84.9% of active devices

I was wondering the same thing. It's been 3+ years and fingerprint authentication has become very widespread. The percentage of those running Lollipop or below is very small.

**Comment by lbalmaceda** [comment]
@rubixhacker  @hon13002 There's support for Fingerprint through one of the `CredentialsManager` classes available in the SDK. You can read more about them in the readme https://github.com/auth0/Auth0.Android/#encryption-enforced-min-api-21.

**Comment by hon13002** [comment]
@lbalmaceda I asked about biometrics in a community post and this is the answer I got from a Community Engineer:

"Unfortunately got bad news for you. I researched through our stack and it seems that the only option we have in terms of biometrics is when using Swift SDK."

However when I was doing research yesterday I did see that bit in the readme. I guess when I get around to actually adding biometrics to both our Android and iOS apps I'll have to try it out. Thanks!

**Comment by lbalmaceda** [comment]
"Biometrics" is too big. I think iOS does support face unlock, but in the case of Android we've only tested it with fingerprint so far. 

This works by delegating the authentication to an external activity/screen that belongs to the Android OS rather than our SDK. It could have support for Face Unlock provided the device is capable of that, but that might also not be the case since I saw @rubixhacker creating https://github.com/auth0/Lock.Android/issues/569.
